testing/e2e/suites/nginxIngressController.go

```diff
@@ -187,6 +204,101 @@ func nicTests(in infra.Provisioned) []test {
 					return err
 				}
 
+				lgr.Info("finished testing")
+				return nil
+			},
+		},
+		{
+			name: "testing DefaultSSLCertificate validity",
+			cfgs: builderFromInfra(in).
+				withOsm(in, false, true).
+				withVersions(manifests.OperatorVersionLatest).
+				withZones([]manifests.DnsZoneCount{manifests.DnsZoneCountNone}, []manifests.DnsZoneCount{manifests.DnsZoneCountNone}).
+				build(),
+			run: func(ctx context.Context, config *rest.Config, operator manifests.OperatorConfig) error {
+				lgr := logger.FromContext(ctx)
+				lgr.Info("starting test")
+
+				c, err := client.New(config, client.Options{
+					Scheme: scheme,
+				})
+				if err != nil {
+					return fmt.Errorf("creating client: %w")
+				}
+
+				// get keyvault uri
+				kvuri := in.Cert.GetId()
+
+				// create defaultSSLCert
+				defaultSSLCert := v1alpha1.DefaultSSLCertificate{
+					KeyVaultURI: &kvuri,
+				}
+
+				// create nic and upsert
+				testNIC := manifests.NewNginxIngressController("nginx-ingress-controller", "nginxingressclass")
+				testNIC.Spec.DefaultSSLCertificate = &defaultSSLCert
+				if err := upsert(ctx, c, testNIC); err != nil {
+					return fmt.Errorf("upserting NIC: %w", err)
+				}
+
+				var service v1alpha1.ManagedObjectReference
+				lgr.Info("waiting for service associated with private NIC to be ready")
+				if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+					lgr.Info("checking if private NIC service is ready")
+					var nic v1alpha1.NginxIngressController
+					if err := c.Get(ctx, client.ObjectKeyFromObject(testNIC), &nic); err != nil {
+						return false, fmt.Errorf("get private nic: %w", err)
+					}
+
+					if nic.Status.ManagedResourceRefs == nil {
+						return false, nil
+					}
+
+					for _, ref := range nic.Status.ManagedResourceRefs {
+						if ref.Kind == "Service" {
+							lgr.Info("found service")
+							service = ref
+							return true, nil
+						}
+					}
+
+					lgr.Info("service not found")
+					return false, nil
+				}); err != nil {
+					return fmt.Errorf("waiting for test NIC to be ready: %w", err)
+				}
+
+				lgr.Info("waiting for spc to be created after NIC")
+				if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+					lgr.Info("checking if associated SPC is created")
+					lgr.Info("checking if spc is created")
+					spc := &secv1.SecretProviderClass{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      keyvault.DefaultNginxCertName(testNIC),
+							Namespace: "app-routing-system",
+						},
+					}
+					cleanSPC := &secv1.SecretProviderClass{}
+
+					if err := c.Get(ctx, client.ObjectKeyFromObject(spc), cleanSPC); err != nil {
+						return false, fmt.Errorf("getting SPC: %w", err)
+					} else {
+						return true, nil
+					}
+
+					lgr.Info("spc not found")
```

Comment:
it's impossible to ever reach this block of code. you have an if else above that always returns. we should remove the else.